scripts/ci/build_emscripten.sh

```diff
@@ -49,17 +49,17 @@ cd $WORKSPACE
 # shellcheck disable=SC2166
 if [[ "$CIRCLE_BRANCH" = release || -n "$CIRCLE_TAG" || -n "$FORCE_RELEASE" || "$(git tag --points-at HEAD 2>/dev/null)" == v* ]]
 then
-	echo -n >prerelease.txt
+	rm cmake/prerelease.lock
 else
 	# Use last commit date rather than build date to avoid ending up with builds for
 	# different platforms having different version strings (and therefore producing different bytecode)
 	# if the CI is triggered just before midnight.
-	TZ=UTC git show --quiet --date="format-local:%Y.%-m.%-d" --format="ci.%cd" >prerelease.txt
-fi
-if [ -n "$CIRCLE_SHA1" ]
-then
-	echo -n "$CIRCLE_SHA1" >commit_hash.txt
+	TZ=UTC git show --quiet --date="format-local:%Y.%-m.%-d" --format="ci.%cd" >cmake/prerelease.lock
 fi
```

Comment:
`prerelease.lock` no longer works like the old `prerelease.txt`. Writing a date there will not make it appear in the version string.

Anyway, please replace this with passing the right flags to the `emcmake cmake` command below.

Reply:
> For https://github.com/ethereum/solidity/pull/12717#discussion_r833473218, are any cmake flags even required ?
> If it's a release, I can just delete the prerelease.lock
> And it's if not a release, I can just leave the prerelease.lock file where it is and the buildinfo.cmake script should take care of generating the earstwhile prerelease string

You should be looking at the other way around. Is `prerelease.lock` even required? If you can achieve it with cmake options, that's a cleaner solution. The only reason we keep the `prerelease.lock` mechanism is that we need it for the archive generated by github.